code/modules/jobs/job_types/security_officer.dm

```diff
@@ -74,55 +76,40 @@ GLOBAL_LIST_EMPTY(security_officer_distribution)
 		// This should theoretically still run if a player isn't in the distributions, but isn't a late join.
 		GLOB.security_officer_distribution[REF(spawning)] = department
 
-	var/ears = null
-	var/accessory = null
-	var/list/dep_trim = null
-	var/destination = null
-
-	switch(department)
-		if(SEC_DEPT_SUPPLY)
-			ears = /obj/item/radio/headset/headset_sec/alt/department/supply
-			dep_trim = /datum/id_trim/job/security_officer/supply
-			destination = /area/security/checkpoint/supply
-			accessory = /obj/item/clothing/accessory/armband/cargo
-		if(SEC_DEPT_ENGINEERING)
-			ears = /obj/item/radio/headset/headset_sec/alt/department/engi
-			dep_trim = /datum/id_trim/job/security_officer/engineering
-			destination = /area/security/checkpoint/engineering
-			accessory = /obj/item/clothing/accessory/armband/engine
-		if(SEC_DEPT_MEDICAL)
-			ears = /obj/item/radio/headset/headset_sec/alt/department/med
-			dep_trim = /datum/id_trim/job/security_officer/medical
-			destination = /area/security/checkpoint/medical
-			accessory = /obj/item/clothing/accessory/armband/medblue
-		if(SEC_DEPT_SCIENCE)
-			ears = /obj/item/radio/headset/headset_sec/alt/department/sci
-			dep_trim = /datum/id_trim/job/security_officer/science
-			destination = /area/security/checkpoint/science
-			accessory = /obj/item/clothing/accessory/armband/science
-
-	if(accessory)
-		var/obj/item/clothing/under/worn_under = spawning.w_uniform
-		worn_under.attach_accessory(new accessory)
-
-	if(ears)
-		if(spawning.ears)
-			qdel(spawning.ears)
-		spawning.equip_to_slot_or_del(new ears(spawning),ITEM_SLOT_EARS)
-
-	// If there's a departmental sec trim to apply to the card, overwrite.
-	if(dep_trim)
-		var/obj/item/card/id/worn_id = spawning.get_idcard(hand_first = FALSE)
-		SSid_access.apply_trim_to_card(worn_id, dep_trim)
-		spawning.sec_hud_set_ID()
-
+	var/list/department_outfit_keys = list(
+		SEC_DEPT_SUPPLY = list(
+			DEPSEC_OUTFIT = /datum/outfit/job/security/supply,
+			DEPSEC_DESTINATION = /area/security/checkpoint/supply
+		),
+		SEC_DEPT_ENGINEERING = list(
+			DEPSEC_OUTFIT = /datum/outfit/job/security/engineering,
+			DEPSEC_DESTINATION = /area/security/checkpoint/engineering
+		),
+		SEC_DEPT_MEDICAL = list(
+			DEPSEC_OUTFIT = /datum/outfit/job/security/medical,
+			DEPSEC_DESTINATION = /area/security/checkpoint/medical
+		),
```

Comment:
wtf

Reply:
Replaced the switch loop with this. Is there an issue?

Reply:
![image](https://user-images.githubusercontent.com/58055496/141087848-e88da8b4-efaa-4add-af10-df867a1cbb64.png)